common/enum_base.h

```diff
@@ -83,6 +83,14 @@ class EnumBase {
   // Prints this value using its name.
   auto Print(llvm::raw_ostream& out) const -> void { out << name(); }
 
+  // TODO: common/ostream.h tries to provide this, but does not reliably work on
+  // EnumBase children. Try to find a more generic solution.
```

Comment:
Yeah, I was thinking it might be something like that, but then why does it work for something like SemIR::Function? I think I just don't understand the lookup well enough...

Reply:
(or why don't other SemIR enums hit the exact same issue?)

Reply:
It also matters where the call to `operator<<` is -- if the call is happening from within namespace `Carbon` or something nested within it, at a point where the templated `operator<<` is visible to unqualified lookup and not hidden by any other `operator<<` in an inner namespace, then that outer operator will be found. So this will depend on whether there are other `operator<<`s in the inner namespace, whether the use of `<<` lives in a template, and if the `<<` is in a template, whether `common/ostream.h` was included before the template was defined.

Our approach here was very fragile, and we're sadly seeing the consequences of that. :(